File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/domain/SingleUserViewModel.java
package com.kalashnyk.denys.samplemvvmapp.domain;

import android.annotation.SuppressLint;
import android.app.Application;
import android.arch.lifecycle.LiveData;

import com.kalashnyk.denys.samplemvvmapp.presentation.widget.SingleLiveEvent;
import com.kalashnyk.denys.samplemvvmapp.repository.AppRepository;
import com.kalashnyk.denys.samplemvvmapp.repository.database.entity.UserEntity;

public class SingleUserViewModel extends BaseViewModel {

    private AppRepository mRepository;
    private SingleLiveEvent<UserEntity> liveDataItems = new SingleLiveEvent<UserEntity>();

    public SingleUserViewModel(Application application, AppRepository repository) {
        super(application);
        mRepository = repository;
    }

    @SuppressLint("CheckResult")
    public void getItem(int id) {
        mRepository.getUser(id).subscribe(userEntity -> liveDataItems.setValue(userEntity), Throwable::printStackTrace);
    }

    public LiveData<UserEntity> getLiveDataItem() {
        return liveDataItems;
    }
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/domain/SingleUserViewModel.kt
package com.kalashnyk.denys.kotlinsample.domain

import android.annotation.SuppressLint
import android.app.Application
import android.arch.lifecycle.LiveData
import com.kalashnyk.denys.kotlinsample.presentation.widget.SingleLiveEvent
import com.kalashnyk.denys.kotlinsample.repository.AppRepository
import com.kalashnyk.denys.kotlinsample.repository.database.entity.UserEntity

class SingleUserViewModel(application: Application, private val mRepository: AppRepository) : BaseViewModel(application) {

    private val liveDataItem = SingleLiveEvent<UserEntity>()

    @SuppressLint("CheckResult")
    fun getItem(id: Int) {
        mRepository.getUser(id).subscribe { list -> liveDataItem.value = list }
    }

    fun getLiveDataItem(): LiveData<UserEntity> {
        return liveDataItem
    }
}